Add warning when prometheus is running in non-privileged mode (#92)
sourcegraph/deploy-sourcegraph-helm: charts/sourcegraph/templates/NOTES.txt

{{- "\n" }}

Thank you for installing {{ .Chart.Name }}.

Your release is named {{ .Release.Name }}.

To learn more about supported configuration, check out https://docs.sourcegraph.com/admin/install/kubernetes/helm.

{{- "\n" }}

{{- if not .Values.prometheus.privileged }}
🚧 Warning 🚧

> You have set 'prometheus.privileged' to 'false'. As a result, our built-in observability stack will not function properly. 

Reviewer: We should include more details here about what won't work so that customers can make an informed choice.

> You have set 'prometheus.privileged' to 'false'. As a result, our built-in observability stack will not function properly. 

Reviewer: ```suggestion
You have set 'prometheus.privileged' to 'false' and Prometheus will run with reduced privileges. As a result, our built-in observability stack will not function properly. 
```

{{- "\n" }}

Thank you for installing {{ .Chart.Name }}.

Your release is named {{ .Release.Name }}.

To learn more about supported configuration, check out https://docs.sourcegraph.com/admin/install/kubernetes/helm.

{{- "\n" }}

{{- if not .Values.prometheus.privileged }}
🚧 Warning 🚧

You have set 'prometheus.privileged' to 'false' and Prometheus will run with reduced privileges.
Our observability stack requires some cluster-wide privileges to display provisioning metrics.

> As a result, our built-in observability stack will not function properly. 

Reviewer: ```suggestion

```

> These metrics will provide critial information to help you scale the Sourcegraph deployment.

Reviewer: ```suggestion
These metrics provide critical information to help you scale the Sourcegraph deployment.
```

> Therefore, please assign proper permission to Service Account '{{ include "sourcegraph.serviceAccountName" (list . "prometheus") }}'.

Reviewer: ```suggestion
Provisioning metrics can be restored by assigning elevated permissions to Service Account '{{ include "sourcegraph.serviceAccountName" (list . "prometheus") }}'.
```

> You can do so by creating the proper ClusterRole and ClusterRoleBinding. 

Reviewer: ```suggestion
```

> You may consult https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/templates/prometheus/prometheus.ClusterRole.yaml

Reviewer: ```suggestion
Consult https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/templates/prometheus/prometheus.ClusterRole.yaml for the necessary permissions.
```

> for a list of expected permissions.

Reviewer: ```suggestion
```
